File: Plugins/Lpp.Dns.General.Metadata/Content/Models/Terms/ClinicalSetting.ts
/// <reference path="../../../../../Lpp.Mvc.Composition/Lpp.Mvc.Boilerplate/jsBootstrap.d.ts" />
/// <reference path="../Terms.ts" />

module RequestCriteriaModels {
    export interface IClinicalSettingsTermData extends RequestCriteriaModels.ITermData {
        ClinicalSetting: ClinicalSettingTypes;
    }

    export enum ClinicalSettingTypes {
        NotSpecified = -1,
        Any = 0,
        InPatient = 1,
        OutPatient = 2,
        Emergency = 3
    }
}

File: Plugins/Lpp.Dns.General.CriteriaGroupCommon/Models/Terms/ClinicalSetting.cs
using System.Runtime.Serialization;

namespace RequestCriteria.Models
{
    [DataContract]
    public class ClinicalSettingData : TermData
    {
        [DataMember]
        public ClinicalSettingTypes ClinicalSetting { get; set; }
    }

    [DataContract]
    public enum ClinicalSettingTypes
    {
        [EnumMember]
        NotSpecified = -1,
        [EnumMember]
        Any = 0,
        [EnumMember]
        InPatient = 1,
        [EnumMember]
        OutPatient = 2,
        [EnumMember]
        Emergency = 3
    }
}
